CRM/Contribute/BAO/ContributionRecur.php

```diff
@@ -835,7 +835,6 @@ public static function getRecurringFields() {
    */
   public static function updateOnNewPayment($recurringContributionID, $paymentStatus, $effectiveDate) {
 
-    $effectiveDate = $effectiveDate ? date('Y-m-d', strtotime($effectiveDate)) : date('Y-m-d');
```

Comment:
This line appears to have come over from one of the PRs I melded in error (mine) and caused problems,

The missing test (which I'll add against master) is that completing a pending contribution using Payment.create/completeOrder correctly updates next_sched_contribution_date